leshan-core/src/main/java/org/eclipse/leshan/core/request/execute/Arguments.java

```diff
@@ -0,0 +1,260 @@
+                String argument = arglist.substring(validBegin, separatorIndex);
+
+                ArgumentParser argumentParser = new ArgumentParser(argument, arglist);
+                if (argumentParser.isValid()) {
+                    argumentList.add(argumentParser.parse());
+                    validBegin = separatorIndex + 1;
+                }
+                beginProbe = separatorIndex + 1;
+            }
+
+            String argument = arglist.substring(beginProbe);
+            ArgumentParser argumentParser = new ArgumentParser(argument, arglist);
+            argumentList.add(argumentParser.parse());
+        }
+
+        return new Arguments(argumentList);
+    }
+
+    private static class ArgumentParser {
+        private final String digitPart;
+        private final String valueDecorated;
+        private final String content;
+
+        public ArgumentParser(String argument, String content) {
+            this.content = content;
+            String[] keyValue = argument.split("=", 2);
+            digitPart = keyValue[0];
+            valueDecorated = keyValue.length == 1 ? null : keyValue[1];
+        }
+
+        private void validate() throws InvalidArgumentException {
+            if (!isValid()) {
+                throw new InvalidArgumentException("Unable to parse Arguments [%s]", content);
+            }
+        }
+
+        private Argument parse() throws InvalidArgumentException {
+            validate();
+            int digit = Integer.parseInt(digitPart);
+            String value = null;
+            if (valueDecorated != null) {
+                value = StringUtils.removeEnd(StringUtils.removeStart(valueDecorated, "'"), "'");
+            }
+
+            return new Argument(digit, value);
+        }
+
+        private boolean isValid() {
+            if (digitPart.length() != 1) {
+                return false;
+            }
+
+            if (valueDecorated != null) {
+                return valueDecorated.length() >= 2 && valueDecorated.charAt(0) == '\''
+                        && valueDecorated.charAt(valueDecorated.length() - 1) == '\'';
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Serialize {$link Arguments} into text representation. In case of empty arguments returns null.
+     */
+    public String serialize() {
+        StringBuilder sb = new StringBuilder();
+
+        for (Argument argument : this) {
+            if (sb.length() > 0) {
+                sb.append(",");
+            }
+
+            sb.append(argument.getDigit());
+
+            if (argument.getValue() != null) {
+                sb.append("='");
+                sb.append(argument.getValue());
+                sb.append("'");
+            }
+        }
+
+        if (sb.toString().length() == 0) {
+            return null;
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * Creates a map that represents arguments. Every argument's null value is converted to empty string "".
```

Comment:
I missed that part at first review :  `Every argument's null value is converted to empty string "".`

I'm not sure we want to do that.

Reply:
I applied this behaviour from LwM2mTestObject.storeParams() - there was ```arguments.put(id, "");``` if argument has no value. But indeed we should keep original value.

Reply:
About `LwM2mTestObject`, I'm not sure why I did that ? :thinking:
We should probably retest this in leshan-server-demo to be sure this works with `null` instead of empty string.

Reply:
After dropping conversion null into "" while executing demo I got ```The validated collection contains an element not of type java.lang.String at index: 1``` exception.

I think that Arguments API should be consistent according to null value, so I just remap value in LwM2mTestObject (client implementation should be responsible for handling null values).